LuaSkin/LuaSkin/Skin.m

```diff
@@ -294,6 +338,15 @@ - (void)checkArgs:(int)firstArg, ... {
                 break;
             case LUA_TNUMBER:
                 lsType = LS_TNUMBER;
+
+                if ((spec & lsType) && (spec & LS_TINTEGER)) {
+                    int isInteger ;
+                    lua_tointegerx(_L, idx, &isInteger) ;
+                    if (!isInteger) {
```

Comment:
@asmagill I am doing a very bad job of explaining myself. In my example above, `9.1` is a valid argument, because the `checkArgs` spec was `LS_TNUMBER | LS_TINTEGER`, which would mean that both floats and integers are acceptable, but afaics the code will reject the floats.